File: src/leetcode/python/easy/SwapNodesInPairs_24.py
class ListNode(object):
    def __init__(self, x):
        self.val = x
        self.next = None


class SwapNodesInPairs_24(object):
    def swapPairs(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        if not head:
            return None
        if not head.next:
            return head
        head.next.next = self.swapPairs(head.next.next)
        h = head.next
        head.next = h.next
        h.next = head
        return h

File: src/leetcode/java/SwapNodesInPairs_24.java
package leetcode.java;

public class SwapNodesInPairs_24 {
  public ListNode swapPairs(ListNode head) {
    if (head == null) return null;
    if (head.next == null) return head;
    head.next.next = swapPairs(head.next.next);
    ListNode h = head.next;
    head.next = h.next;
    h.next = head;
    return h;
  }
}

class ListNode {
  int val;
  ListNode next;
  ListNode(int x) { val = x; }
}